src/main/kotlin/com/criptext/mail/scenes/emaildetail/workers/UpdateUnreadStatusWorker.kt

```diff
@@ -22,23 +25,40 @@ class UpdateUnreadStatusWorker(
         private val db: EmailDetailLocalDB,
         private val threadId: String,
         private val updateUnreadStatus: Boolean,
+        httpClient: HttpClient,
+        activeAccount: ActiveAccount,
         private val currentLabel: Label,
         override val publishFn: (EmailDetailResult.UpdateUnreadStatus) -> Unit)
     : BackgroundWorker<EmailDetailResult.UpdateUnreadStatus> {
 
+    private val apiClient = EmailDetailAPIClient(httpClient, activeAccount.jwt)
+
     override val canBeParallelized = false
 
     override fun catchException(ex: Exception): EmailDetailResult.UpdateUnreadStatus {
         return EmailDetailResult.UpdateUnreadStatus.Failure()
     }
 
-    override fun work(reporter: ProgressReporter<EmailDetailResult.UpdateUnreadStatus>): EmailDetailResult.UpdateUnreadStatus? {
+    private fun updateUnreadEmailStatus(){
         val rejectedLabels = Label.defaultItems.rejectedLabelsByMailbox(currentLabel).map { it.id }
         val emailIds = db.getFullEmailsFromThreadId(threadId, rejectedLabels).map {
             it.email.id
         }
         db.updateUnreadStatus(emailIds, updateUnreadStatus)
-        return EmailDetailResult.UpdateUnreadStatus.Success(threadId, updateUnreadStatus)
+    }
+
+    override fun work(reporter: ProgressReporter<EmailDetailResult.UpdateUnreadStatus>): EmailDetailResult.UpdateUnreadStatus? {
+        val result = Result.of { apiClient.postThreadReadChangedEvent(listOf(threadId),
+                updateUnreadStatus)}
+                .map { updateUnreadEmailStatus() }
```

Comment:
Here's a little `Result` gotcha. `map` merely uses the provided function to transform type `V` into `U` **without any error handling**. If you want exceptions to be captured, your code should run inside a `Result.of {}` and in order to chain that, you have to use `flatMap` instead·